security/fma-fjord.md

```diff
@@ -131,9 +164,21 @@ For more detail, feel free to review: https://github.com/ethereum-optimism/optim
 
 - **Recovery Path(s)**: If this turned out to be a major problem (very unlikely), we can always revert back to zlib compression, which should be closer to FastLZ and then not exhibit this attack surface. If it only affects a small number of transactions, we can ship an improvement to the cost function in the next hardfork and live with a few underpriced transactions in the meantime.
 
-## Generic items we need to take into account:
+### P256VERIFY is too expensive for Fault Proofs
 
-### Chain Halt at Activation
+- **Description:** The introduction of new precompiles always carries a risk of making it too costly for the op-challenger to generate execution traces and fault proofs that contain the precompile.
+
+- **Risk Assessment:** Low. [Based on Cannon performance testing of various precompiles](https://www.notion.so/865e2b86c24a404dbf6f2aa35b94a0eb?pvs=21), including `P256VERIFY`, it’s not costly for offchain Cannon to generate fault proofs. The gas schedule correctly reflects the offchain costs of `P256VERIFY` traces*.* There aren’t known adversarial inputs to `P256VERIFY` that would induce an unexpectedly long Cannon execution trace.
+
+- **Mitigations:** [Cannon precompile performance testing](https://www.notion.so/865e2b86c24a404dbf6f2aa35b94a0eb?pvs=21) show that it’s relatively, compared to existing precompiles, inexpensive to generate traces of `P256VERIFY` operations.
+
+- **Detection:** The op-challenger will be slower in responding to games and may ultimately lose dispute games. op-dispute-mon contains metrics that are configured to page proofs-squad and security on either case.
+
+- **Recovery Path(s)**:
```

Comment:
@inphi can you provide a code suggestion we can apply to fill this out?

Reply:
```suggestion
- **Recovery Path(s)**:

Several recovery options are available, depending on the severity of the problem. These options, listed in order of increasing severity, include:

1. Blacklisting problematic games, if and only if the issue affects a limited number of games.
Increasing the op-challenger compute resources. Generating such traces is CPU-bound and may be alleviated with more computational power.
1. If the above measures prove ineffective, upgrading the system to use the `PermissionedDisputeGame` as the respected game type. This should be implemented if there's sufficient time for the Guardian to execute the upgrade.
1. As a last resort, if time is critically limited, executing the pre-signed pause.

Each option should be considered carefully, with the least disruptive solution attempted first before escalating to more severe measures. Ultimately, the pre-signed pause must be executed if there is not ample time available for the other recovery options.
```